### ISSUE TITLE
Error:Adding OData Connected Service to the project failed: The 'Name' attribute is invalid

### ISSUE THREAD

**Issue by bydamien** [OData/lab#89]
hello i am getting this error message when i try add xamarin forms application
Adding Nuget Packages
Generating Client Proxy ...
Error:Adding OData Connected Service to the project failed: The 'Name' attribute is invalid - The value 'ErpWeb365.ApiService' is invalid according to its datatype 'http://schemas.microsoft.com/ado/2007/05/edm:TSimpleIdentifier' - The Pattern constraint failed.

**Comment by bydamien** [comment]
nobody wants to help me ?

**Comment by unchase** [comment]
I tried it in my own service. All good.
You can try my [Unchase ODataConnectedService](https://marketplace.visualstudio.com/items?itemName=Unchase.unchaseodataconnectedservice) for solve this problem.

**Comment by gtrennert** [comment]
Hello unchase,
I have this error when trying to create an odata connected service to web service on page 21 (business central) : 
Generating Client Proxy for OData V3...
Erreur :Échec de l'ajout de Unchase OData Connected Service au projet : Le schéma spécifié n'est pas valide. Erreurs : 
(0,0) : erreur 0005: L'attribut 'Name' n'est pas valide - La valeur '_Balance_LCY_Payment_in_progress_LCY' n'est pas valide selon son type de données 'http://schemas.microsoft.com/ado/2007/05/edm:TSimpleIdentifier' - Échec de la contrainte Pattern.
Any Idea ?

**Comment by unchase** [comment]
@gtrennert, сould you send the metadata file (or public URI) for the service to be added so that I can check the issue?
You can send it to me by email (see in my github profile).

**Comment by unchase** [comment]
@gtrennert. OData library does not support a naming in form of "\_Balance_LCY_Payment_in_progress_LCY", and the Connected service too.
If you can, try to rename this parameter without "_" in the beginning.

P.S. I removed the first "_" in "\_Balance_LCY_Payment_in_progress_LCY" (in metadata file) => "Balance_LCY_Payment_in_progress_LCY" and it works!

**Comment by gtrennert** [comment]
I understand what you are saying, but I'm just testing web service access for Business central, its microsoft standard and I just don't understand why there is a leading underscore.
Web services in business central should work without any modification of standard tables (modifications which are no prohibited in this product)

**Comment by unchase** [comment]
I agree. In any case, the client code generator cannot correctly handle this naming.

**Comment by gtrennert** [comment]
I will create a new page in which I will exclude all this fields beginning with _
Thanks a lot for your answers

**Comment by unchase** [comment]
@gtrennert, you're welcome.

**Comment by nick-webbgh** [comment]
In Visual Studio 2017, on trying to add an OData Connected Service, I get the error below:-
Adding OData Connected Service to the project failed: The current schema element does not support text 'NAV.NavType/Real'. Any thoughts?

**Comment by unchase** [comment]
@nick-webbgh, сould you send the metadata file to me by email? I will check the issue.

**Comment by nick-webbgh** [comment]
Please see attached. Thanks in advance for your help.

**Comment by unchase** [comment]
There is no attached files.

**Comment by unchase** [comment]
Hi. There is no attached files.
Can you send again to spiritkola@hotmail.com?

**Comment by nick-webbgh** [comment]
I did attach the file last time. Maybe it got blocked. I’ve zipped it this time.

**Comment by gtrennert** [comment]
Hello again unchase,
I have the following issue - when configuring a metadata endpoint I get the following type of messages : see picture joined
![02 message](https://user-images.githubusercontent.com/2358075/65312095-1aa15680-db92-11e9-9096-017a6d0bbe10.jpg)
As I may have many web services published with the path "https://bc190709:7048/NAV/OData/$metadata" I get many of this messages and after confirming the first ones I get errors. Is there a way to avoid this messages ?
I also tried to enter longer urls like https://bc190709:7048/NAV/OData/Company('CRONUS%20FR')/TestOdata but then I get error message : 
![03 cannot access](https://user-images.githubusercontent.com/2358075/65319966-23e5ef80-dba1-11e9-891a-36bf9494c38c.jpg)

**Comment by gtrennert** [comment]
Here a better screenshot of all the messages : 
![04 messages multiple](https://user-images.githubusercontent.com/2358075/65324169-bab7a980-dbab-11e9-8f2a-99b491a70151.jpg)

**Comment by gtrennert** [comment]
even now that i deleted all other published webservices i have many popups like mentionned above

and second thing I wanted to ask : Is there any place with help - because I am a little bit lost in c# howto go for inserting data via web service

**Comment by gtrennert** [comment]
Because I tried to find help but there is always something missing - for example : 
            DataServiceResponse response = nav.SaveChanges();
'NAV' ne contient pas de définition pour 'SaveChanges' et aucune méthode d'extension accessible 'SaveChanges' acceptant un premier argument de type 'NAV' n'a été localisée (une directive using ou une référence d'assembly est-elle manquante ?)

**Comment by unchase** [comment]
@gtrennert , could you create an issue in my repository: [ask a question](https://marketplace.visualstudio.com/items?itemName=unchase.unchaseODataConnectedService&ssr=false#qna) or [report a bug](https://github.com/unchase/Unchase.OData.Connectedservice/issues/new?title=BUG)?
It's easier for me to track it.

Check the issue after updating to [Unchase OData Connected Service v1.3.0](https://marketplace.visualstudio.com/items?itemName=Unchase.unchaseodataconnectedservice)

On the last question: I can advise [official documentation](https://docs.microsoft.com/ru-ru/odata/client/basic-crud-operations).

**Comment by unchase** [comment]
> Here a better screenshot of all the messages :
> ![04 messages multiple](https://user-images.githubusercontent.com/2358075/65324169-bab7a980-dbab-11e9-8f2a-99b491a70151.jpg)

Fixed in [Unchase OData Connected Service v1.3.2](https://marketplace.visualstudio.com/items?itemName=Unchase.unchaseodataconnectedservice).

**Comment by ghost** [comment]
Hello unchase - I am trying to use  Unchase OData Connected Service 1.4.3 to connect to Accumatica OData feed but am getting this same invalid name issue - 'The value 'DB-StorageDetails' is invalid according to its datatype.... .... The Pattern constraint failed.
Any idea why this would be happening?
![image](https://user-images.githubusercontent.com/91299198/134607195-b6a65a43-fdf9-4095-b1cf-a9f06fe1d366.png)

**Comment by nick-webbgh** [comment]
The first thing I would try is to remove the dash from DB-StorageDetails?

**Comment by ghost** [comment]
Thanks for your reply, the dash in all the names does appear to be the issue but these are all the default objects in Accumatica OData feed and they work fine when consumed in other OData applications.
I'll have to see if our admin can have them renamed

**Comment by nick-webbgh** [comment]
Just out of interest, have you tried using the ordinary OData Connected Service instead of the Unchase one?

**Comment by ghost** [comment]
Yes I did try it first but had issues with authentication which was easily solved using the Unchase one

**Comment by unchase** [comment]
@zdrehs What kind of issues did you have exactly?

**Comment by nick-webbgh** [comment]
You can add authentication details into the custom headers (see link below):-

https://docs.microsoft.com/en-us/odata/connectedservice/accessing-endpoints-with-auth

**Comment by baguilar-work** [comment]
Ive been reading through the comments and from what I can tell my issue should not be an error: `Adding Unchase OData Connected Service to the project failed: The attribute 'Precision' was not expected in the given context.`

However, I cant seem to add it through VS 2022 using the newest version (1.5.1) of Unchase OData Service found through VS.

I had to upload the file im using as a .txt, as a heads up. Since it did not allow .xml

Any help would be greatful

Thanks!

[imlsMetaData.txt](https://github.com/OData/lab/files/9471430/imlsMetaData.txt)
